### DISCUSSION TITLE
First taughts

### DISCUSSION THREAD

**Discussion by achaayb** [pocketbase/pocketbase#424] [Q&A]
Hello, at first i'd like to congratulate you on this amazing project hype. i find the idea really interesting since most projects only require crud-ish stuff, here are some questions
- the subscribe thingie, is a websocket connection established even tho the user isnt allowed ?
- does pocketbase squeeze every core or is it single core single threaded
- PLEASE keep the minimal bundle minimalistic with a modules option, dont ruin the whole point of this project
- can the project at its current state be used in prod?

**Comment by ganigeorgiev** [comment]
> the subscribe thingie, is a websocket connection established even tho the user isnt allowed ?

PocketBase is using Server-sent events (SSE) for the realtime subscriptions which are just regular long-lived GET HTTP requests.
Access checks are performed when subscribing to a collection or record, respsectively the list or view rules are checked when broadcasting messages (more info you could find in https://pocketbase.io/docs/api-realtime/).
Additionally there is [`OnRealtimeConnectRequest` event hook](https://pocketbase.io/docs/event-hooks/#realtime-hooks) if you want to further restrict the access to the realtime connect endpoint.

---

> does pocketbase squeeze every core or is it single core single threaded

PocketBase is built on top of the standard Go `net/http` package wich serves each request in its own go routine. The number of used cores is determined by the go scheduler (you can further tune it via the [`GOMAXPROCS` env variable](https://pkg.go.dev/runtime#GOMAXPROCS) which defaults to the number of cpu cores available).

---

> can the project at its current state be used in prod?

While I try to keep breaking changes to a minimum, it is still in its early development stage and backward compatibility is not fully guaranteed before reaching v1.0.0. For example, for the next v0.8.0 release there are some planned changes that will require manual migration steps in attempt to simplify the users management per #376.
